packages/cloud_firestore/cloud_firestore_web/lib/field_value_web.dart

```diff
@@ -0,0 +1,28 @@
+part of cloud_firestore_web;
+
+class FieldValueWeb implements FieldValueInterface, web.FieldValue {
+  static Map<String, dynamic> _serverDelegates(Map<String, dynamic> data) {
+    Map<String, dynamic> output = Map.from(data);
+    output.updateAll((key, value) {
```

Comment:
Does this need to check for a Map or Array value? (Again, looking at the jsify implementation to see what type of data we might expect to pass through the function)

```dart
if (value is Map) {
  return _serverDelegates(value);
}
```

Reply:
Wouldn't Map and Array be ordinary dart types that jsify can handle ? or do you mean nested `FieldValue` inside?

Reply:
Yes, what if you have a nested FieldValue inside of an array/map contained within the data? I think this method needs to traverse the whole "tree" of data, and not just the first level of keys of the map.

Reply:
I've updated it now to handle array and maps.